tutorials/07-voting/README.md

```diff
@@ -0,0 +1,645 @@
+        /// using their `Receiver` reference.
+        /// We say `&AnyResource{Receiver}` to say that the recipient can be any resource
+        /// as long as it implements the Receiver interface
+        pub fun mintTokens(amount: UFix64, recipient: Capability<&AnyResource{FungibleToken.Receiver}>) {
+            let recipientRef = recipient.borrow()
+                ?? panic("Could not borrow a receiver reference to the vault")
+
+            VotingTutorialGovernanceToken.totalSupply = VotingTutorialGovernanceToken.totalSupply + UFix64(amount)
+            recipientRef.deposit(from: <-create Vault(balance: amount))
+
+            emit TokensMinted(amount: amount)
+        }
+    }
+
+    /// The init function for the contract. All fields in the contract must
+    /// be initialized at deployment. This is just an example of what
+    /// an implementation could do in the init function. The numbers are arbitrary.
+    init() {
+        self.totalSupply = 0.0
+
+        // Assign paths
+        self.VaultStoragePath = /storage/CadenceVotingTutorialVotingTutorialGovernanceTokenVaultStoragePath
+        self.MinterStoragePath = /storage/CadenceVotingTutorialVotingTutorialGovernanceTokenMinterStoragePath
+        self.VaultPublicPath = /public/CadenceVotingTutorialVotingTutorialGovernanceTokenVaultPublicPath
+
+        /// Create the Vault with the initial balance and put it in account storage.
+        /// account.save saves an object to the specified `to` path
+        /// The path is a literal path that consists of a domain and identifier
+        /// The domain must be `storage`, `private`, or `public`
+        /// the identifier can be any name
+        let vault <- create Vault(balance: self.totalSupply)
+        self.account.save(<-vault, to: self.VaultStoragePath)
+
+        /// Create a public capability to the stored Vault that exposes VotingWeight
+        self.account.link<&VotingTutorialGovernanceToken.Vault{FungibleToken.Provider, FungibleToken.Receiver, FungibleToken.Balance, VotingTutorialGovernanceToken.VotingWeight}>(VotingTutorialGovernanceToken.VaultPublicPath, target: VotingTutorialGovernanceToken.VaultStoragePath)
+
+        /// Create a new Minter resource and store it in account storage
+        self.account.save(<-create VaultMinter(), to: self.MinterStoragePath)
+
+        /// Emit initialization event
+        emit TokensInitialized(initialSupply: self.totalSupply)
+    }
+}
+```
+
+`VotingTutorialAdministration` contains a struct `ProposalData` which is used to store both the proposal and the total votes.
+Then we have two resources, a `Ballot` and an `Administrator` resource.
+A user can request a `Ballot`, and then vote for a proposal, effectively tallying the vote weight in this contract.
+Using a resource type is logical for this application, because if a user wants to delegate their vote, they can send that `Ballot` to another account.
+Access to the `Administrator` resource is needed in order to add proposals.
```

Comment:
Same comment applies here that I made about the token contract. We need a whole long section here that goes into detail about each part of the contract and explains each feature with code samples.

Reply:
I have now included one code snippet. More snippets would result pretty quick in pasting the whole contract code I think. Instead, I resorted to describe the various components in more detail.